database.cc

```diff
@@ -481,6 +486,11 @@ database::setup_metrics() {
                                        " to be able to admit new ones, if there is a shortage of permits."),
                        {streaming_label_instance}),
 
+        sm::make_derive("reads_shed_due_to_overload", _read_concurrency_sem.get_stats().total_reads_shed_due_to_overload,
+                       sm::description("The number of reads shed because the admission queue reached its max capacity."
+                                       " When the queue is full, excessive reads are shed to avoid overload."
+                       {streaming_label_instance}),
+
```

Comment:
Unrelated: looks like there is duplication, we can extract the function to create metrics, and call it twice.